mongo_connector/connector.py

```diff
@@ -732,84 +741,76 @@ def apply_exclude_fields(option, cli_values):
+                                   "Include namespace set")
+        if option.value["exclude"]:
+            validate_no_duplicates(option.value["exclude"],
+                                   "Exclude namespace set")
+        if option.value["gridfs"]:
+            validate_no_duplicates(option.value["gridfs"],
+                                   "GridFS namespace set")
+        if option.value["mapping"]:
+            validate_no_duplicates(list(option.value["mapping"].values()),
+                                   "Destination namespace set")
 
-        ns_set = option.value['include']
-        if len(ns_set) != len(set(ns_set)):
-            raise errors.InvalidConfiguration(
-                "Namespace set should not contain any duplicates.")
+    def apply_namespaces(option, cli_values):
+        if (option.value['include'] or option.value['exclude'] or
+                option.value['mapping'] or option.value['gridfs']):
+            return apply_disjoint_namespace_options(option, cli_values)
+        else:
+            return apply_unified_namespace_options(option, cli_values)
 
-        ex_ns_set = option.value['exclude']
-        if len(ex_ns_set) != len(set(ex_ns_set)):
-            raise errors.InvalidConfiguration(
-                "Exclude namespace set should not contain any duplicates.")
+    def apply_unified_namespace_options(option, cli_values):
+        merge_namespaces_cli(option, cli_values)
+        namespace_options = copy.deepcopy(option.value)
+        ns_set = namespace_options.pop('include', None)
+        ex_ns_set = namespace_options.pop('exclude', None)
+        gridfs_set = namespace_options.pop('gridfs', None)
+        dest_mapping = namespace_options.pop('mapping', None)
+        option.value["namespace_options"] = namespace_options
 
-        # not allow to exist both 'include' and 'exclude'
-        if ns_set and ex_ns_set:
-            raise errors.InvalidConfiguration(
-                "Cannot use both namespace 'include' "
-                "(--namespace-set) and 'exclude' "
-                "(--exclude-namespace-set).")
+        validate_namespace_options(
+            namespace_set=ns_set, ex_namespace_set=ex_ns_set,
+            dest_mapping=dest_mapping,
+            namespace_options=namespace_options, gridfs_set=gridfs_set)
 
-        # validate 'include' format
-        for ns in ns_set:
-            if ns.count("*") > 1:
-                raise errors.InvalidConfiguration(
-                    "Namespace set should be plain text "
-                    "e.g. foo.bar or only contains one wildcard, e.g. foo.* .")
-
-        # validate 'exclude' format
-        for ens in ex_ns_set:
-            if ens.count("*") > 1:
-                raise errors.InvalidConfiguration(
-                    "Exclude namespace set should be plain text "
-                    "e.g. foo.bar or only contains one wildcard, e.g. foo.* .")
+    def apply_disjoint_namespace_options(option, cli_values):
```

Comment:
We may also want to log a warning that the old-style namespace options will be going away eventually, since supporting both styles will be difficult to maintain indefinitely.